_include/Monkey.h

```diff
@@ -0,0 +1,38 @@
+#ifndef MONKEY_H
+#define MONKEY_H
+#include "Animal.h"
+
+
+/**
+ * A subclass of the `Animal` class representing a monkey with a name, a location, and movement capabilities.
+ * The monkey moves either vertically or horizantlly, with an option of one or two indexes each step.
+ * New direction and step size drawn every five steps.
+ */
+class Monkey :public Animal {
+
+	int horizontal_direction;	// The current horizontal direction of the monkey (either -1, or 1).
```

Comment:
Not all animals have the same moving capabilities and directions, so i didn't think the members were general enough to include in the parent Animal class.

Reply:
it's a good question, what about the functions turn vertically & horizontally ?, in this class you don't need the turn vertically function, but still it's declared in the animal class, so aren't the members horizontal_direction & vertical_direction the same ?

Reply:
also, with right code you can use only 2 members vertical & horizontal, you don't need the diagonal member, since vertical+ horizontal+ is upper right, vertical- & horizontal- is down left , and go on

Reply:
Another approach you can take is to create an intermediate class, let's call it "DiagonalAnimal," specifically for animals that move in a diagonal direction. By defining the relevant members and behaviors in the DiagonalAnimal class, you can avoid duplicating those declarations in each derived class that exhibits diagonal movement.